lib/proto/proto_conn.c

```diff
@@ -191,23 +191,24 @@ proto_conn_drop(void * conn_cookie, int reason)
 }
 
 /**
- * proto_conn_create(s, sas, decr, nopfs, requirepfs, nokeepalive, K, timeo,
- *     callback_dead, cookie):
+ * proto_conn_create(s, sas, sa_b, decr, nopfs, requirepfs, nokeepalive, K,
+ *     timeo, callback_dead, cookie):
  * Create a connection with one end at ${s} and the other end connecting to
- * the target addresses ${sas}.  If ${decr} is 0, encrypt the outgoing data;
- * if ${decr} is nonzero, decrypt the incoming data.  If ${nopfs} is non-zero,
- * don't use perfect forward secrecy.  If ${requirepfs} is non-zero, drop
- * the connection if the other end tries to disable perfect forward secrecy.
- * Enable transport layer keep-alives (if applicable) on both sockets if and
- * only if ${nokeepalive} is zero.  Drop the connection if the handshake or
+ * the target addresses ${sas}.  Optionally bind to address ${sa_b}.  If
+ * ${decr} is 0, encrypt the outgoing data;  * if ${decr} is nonzero, decrypt
+ * the incoming data.  If ${nopfs} is non-zero, * don't use perfect forward
+ * secrecy.  If ${requirepfs} is non-zero, drop  * the connection if the
+ * other end tries to disable perfect forward secrecy.  Enable transport
+ * layer keep-alives (if applicable) on both sockets if and  only if
+ * ${nokeepalive} is zero.  Drop the connection if the handshake or
  * connecting to the target takes more than ${timeo} seconds.  When the
  * connection is dropped, invoke ${callback_dead}(${cookie}).  Free ${sas}
  * once it is no longer needed.  Return a cookie which can be passed to
  * proto_conn_drop().  If there is a connection error after this
  * function returns, close ${s}.
  */
 void *
-proto_conn_create(int s, struct sock_addr ** sas, int decr, int nopfs,
+proto_conn_create(int s, struct sock_addr ** sas, struct sock_addr * sa_b, int decr, int nopfs,
```

Comment:
Looks like sa_b is not actually used in proto_conn_create?

Reply:
Sorry, forgot - fixed.

Reply:
This should be const.

Reply:
Fixed.